### PR TITLE
[VTA][TSIM] add virtual memory support to tsim example

### PR DESCRIPTION
**Pull request by vegaluisjose**
Hey @liangfu @tmoreau89 ,

This PR fixes TSIM app example, it uses virtual memory now. The reason for adding vm to the app is that the [dpi module now uses vm for memory requests](https://github.com/dmlc/tvm/blob/master/vta/src/dpi/module.cc#L185).

### CODE REVIEW

File: vta/apps/tsim_example/python/tsim.py [hunk 1]
```diff
@@ -21,15 +21,24 @@
 from sys import platform
 
 def get_ext():
+    """Return shared library extension"""
     return ".dylib" if platform == "darwin" else ".so"
 
 def load_dll(dll):
+    """Load shared library
+
+     Parameters
```
Comment (liangfu) [inline vta/apps/tsim_example/python/tsim.py]: Please mind the indentation, leave four spaces before `Parameters` to align with `"""`

File: vta/apps/tsim_example/python/tsim.py [hunk 2]
```diff
@@ -21,15 +21,24 @@
...
+     ------------
+     dll : str
+        Path for shared library
+    """
```
Comment (liangfu) [inline vta/apps/tsim_example/python/tsim.py]: Add `Returns` as well, see a typical style [here](https://github.com/dmlc/tvm/blob/master/python/tvm/build_module.py#L253).

File: vta/apps/tsim_example/CMakeLists.txt [hunk 3]
```diff
@@ -35,8 +35,12 @@ if (CMAKE_CXX_COMPILER_ID MATCHES "GNU" AND
 endif()
 
 file(GLOB TSIM_SW_SRC src/driver.cc)
+list(APPEND TSIM_SW_SRC ${VTA_DIR}/src/vmem/virtual_memory.cc)
+list(APPEND TSIM_SW_SRC ${VTA_DIR}/src/dpi/module.cc)
+
 add_library(sw SHARED ${TSIM_SW_SRC})
-target_include_directories(sw PRIVATE ${VTA_DIR}/include)
+target_compile_definitions(sw PUBLIC USE_TSIM)
```
Comment (liangfu) [inline vta/apps/tsim_example/CMakeLists.txt]: Just out of curiosity, I've seen two places using this macro, one is to define `vta_phy_addr_t`, the other is to define `VTADeviceRun`. What is the reason for adding this in `tsim_example`?

File: vta/apps/tsim_example/python/tsim.py (hunk 2 above)
Comment (vegaluisjose) [reply vta/apps/tsim_example/python/tsim.py]: Fixed

File: vta/apps/tsim_example/python/tsim.py (hunk 1 above)
Comment (vegaluisjose) [reply vta/apps/tsim_example/python/tsim.py]: Fixed

File: vta/apps/tsim_example/CMakeLists.txt (hunk 3 above)
Comment (vegaluisjose) [reply vta/apps/tsim_example/CMakeLists.txt]: You are right, now that we have the virtual memory manager we don't need this for simulation. I am going to remove it from the example.

`USE_TSIM` is basically a quickfix for FPGAs with 64-bit addresses, although I don't see any reason why not using virtual memory manager on actual devices and getting rid of this.

What is your take on using vmem on actual devices and not just in simulation?

Comment (liangfu) [reply vta/apps/tsim_example/CMakeLists.txt]: I think using vmem on actual hardware might drop the performance. Instead, implementing a `MMU` to replace the software vmem would be a better choice.

**Pull request merged**